File: app/src/main/java/br/com/alura/ceep/ui/coffemachine/CoffeAdapter.java
package br.com.alura.ceep.ui.coffemachine;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.AdapterView.OnItemClickListener;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.jetbrains.annotations.NotNull;

public class CoffeAdapter extends RecyclerView.Adapter<CoffeAdapter.ViewHolder> {

    CoffeMachineData[] coffeMachineData;
    Context context;


    public CoffeAdapter (CoffeMachineData[] coffeMachineData, MainActivity activity) {
        this.coffeMachineData = coffeMachineData;
        this.context = activity;
    }

    @NonNull
    @NotNull
    @Override
    public ViewHolder onCreateViewHolder (@NonNull @NotNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from (parent.getContext ());
        View view = layoutInflater.inflate (R.layout.coffe_item_list, parent, false);
        ViewHolder viewHolder = new ViewHolder (view);
        return viewHolder;
    }

    public void onBindViewHolder (@NonNull @NotNull ViewHolder holder, int position) {

        final CoffeMachineData coffeMachineDataList = coffeMachineData[position];
        holder.textViewDescription.setText (coffeMachineDataList.getCoffeDescription ());
        holder.textViewSize.setText (coffeMachineDataList.getCoffeSize ());
        holder.textQtd.setText (coffeMachineDataList.getCoffeSizeTitule ());
        holder.textViewIntensity.setText (coffeMachineDataList.getCoffeIntensity ());
        holder.textInt.setText (coffeMachineDataList.getCoffeIntensityTitule ());
        holder.textViewType.setText (coffeMachineDataList.getCoffeType ());
        holder.coffeImage.setImageResource (coffeMachineDataList.getCoffeImage ());
    }

    @Override
    public int getItemCount () {
        return coffeMachineData.length;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        ImageView coffeImage;
        TextView textViewDescription;
        TextView textViewType;
        TextView textViewSize;
        TextView textViewIntensity;
        TextView textQtd;
        TextView textInt;

        public ViewHolder (@NonNull @NotNull View itemView) {

            super (itemView);
            itemView.setOnClickListener (new View.OnClickListener () {
                @Override
                public void onClick (View v) {
                    itemView.getContext ().startActivity (new Intent (itemView.getContext (), DetailActivity.class));
                }
            });
            coffeImage = itemView.findViewById (R.id.coffeImagetype);
            textViewDescription = itemView.findViewById (R.id.CoffeDescription);
            textViewType = itemView.findViewById (R.id.CoffeType);
            textViewSize = itemView.findViewById (R.id.CoffeSize);
            textViewIntensity = itemView.findViewById (R.id.NumberIntensity);
            textQtd = itemView.findViewById (R.id.QtdCoffe);
            textInt = itemView.findViewById (R.id.CoffeIntensity);
        }
    }

}

File: app/src/main/java/br/com/alura/ceep/ui/coffemachine/presentation/custom/CoffeAdapter.kt
package br.com.alura.ceep.ui.coffemachine.presentation.custom

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.activity.viewModels
import androidx.core.content.ContentProviderCompat.requireContext
import androidx.fragment.app.viewModels
import androidx.recyclerview.widget.RecyclerView
import br.com.alura.ceep.ui.coffemachine.R
import br.com.alura.ceep.ui.coffemachine.domain.Coffee
import br.com.alura.ceep.ui.coffemachine.helpers.CoffesRoomDataBase
import br.com.alura.ceep.ui.coffemachine.presentation.DetailActivity
import br.com.alura.ceep.ui.coffemachine.repository.CoffesRepository
import br.com.alura.ceep.ui.coffemachine.viewmodel.CoffesViewModel

class CoffeAdapter(val selected: (coffee: Coffee) -> Unit) :
    RecyclerView.Adapter<CoffeeViewHolder>() {
    var list = ArrayList<Coffee>()
    private var context: Context? = null


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CoffeeViewHolder {
        context = parent.context
        val layoutInflater = LayoutInflater.from(context)
        val view = layoutInflater.inflate(R.layout.coffe_item_list, parent, false)
        return CoffeeViewHolder(view)
    }

    override fun onBindViewHolder(holder: CoffeeViewHolder, position: Int) {
        holder.bind(list.get(position))
        holder.itemView.setOnClickListener { v: View? ->
            selected(list.get(position))
            val intent = Intent(context, DetailActivity::class.java)
            context?.startActivity(intent)
        }

    }

    override fun getItemCount(): Int {
        return list.count()
    }
}